Composer/packages/ui-plugins/virtual-assistant-creation/src/components/customizeBotPage.tsx

```diff
@@ -0,0 +1,130 @@
+/* eslint-disable format-message/literal-pattern */
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx, css } from '@emotion/core';
+import formatMessage from 'format-message';
+import React, { useContext } from 'react';
+import { RouteComponentProps } from '@reach/router';
+import { DialogWrapper, DialogTypes } from '@bfc/ui-shared';
+import { Dropdown, IDropdownStyles, IDropdownOption } from 'office-ui-fabric-react/lib/Dropdown';
+import { Label } from 'office-ui-fabric-react/lib/Label';
+import { TextField } from 'office-ui-fabric-react/lib/TextField';
+import { Checkbox } from 'office-ui-fabric-react/lib/Checkbox';
+import { Stack } from 'office-ui-fabric-react/lib/Stack';
+
+import { RouterPaths } from '../shared/constants';
+import { AvailablePersonalities } from '../models/creationOptions';
+
+import { AppContext } from './VirtualAssistantCreationModal';
+import { DialogFooterWrapper } from './dialogFooterWrapper';
+
+// -------------------- Styles -------------------- //
+const textFieldStyling = css`
+  width: 400px !important;
+`;
+
+const horizontalStackTokens = { childrenGap: 10 };
+const verticalStackTokens = { childrenGap: 15 };
+
+const dropdownStyles: Partial<IDropdownStyles> = {
+  dropdown: { width: 300 },
+};
+
+// -------------------- CustomizeBotPage -------------------- //
+type CustomizeBotPageProps = {
+  onDismiss: () => void;
+} & RouteComponentProps<{}>;
+
+export const CustomizeBotPage: React.FC<CustomizeBotPageProps> = (props) => {
+  const { state, setState } = useContext(AppContext);
```

Comment:
Are you using the single large context object rather than the recoil state here?
@a-b-r-o-w-n How should UI plug-ins wire plug-in wide state or integrate with app state?

Reply:
I believe we want to limit the visibility a plug in has to recoil state via the shellAPI so plugins should not be able to access recoil state directly. 
The UI plug in for creation implements its own state management to handle state within its experience as any UI plug in would. I believe client should largely not care what a ui plugin uses for its own state management as long as it follows a strict contract when interacting with client state via the ShellAPI. @a-b-r-o-w-n  is that correct? 

Granted the state management I have chosen is resulting in unnecessary rerenders as your next comment indicates so I will fix that.

Reply:
I agree with @pavolum. Plugins will be able to access state through the shell api. Composer internal state management should not leak into plugins.

And I also agree that plugins can bring whatever state management library they prefer.